mat/consts.go

```diff
@@ -31,6 +31,12 @@ const (
 	//  A = U * Σ * V^T
 	// where U is of size m×m, Σ is an m×n diagonal matrix, and V is an n×n matrix.
 	SVDFull
```

Comment:
This is starting to look to me like it should be a bit flag set.

```
const (
	// SVDU computes the thin singular vectors as described for SVDThin, but
	// only computes U~ and Σ, not V~^T.
	SVDU SVDKind = 1 << iota

	// SVDV computes the thin singular vectors as described for SVDThin, but
	// only computes Σ and V~^T, not U.
	SVDV

	// SVDThin computes the thin singular vectors, that is, it computes
	//  A = U~ * Σ * V~^T
	// where U~ is of size m×min(m,n), Σ is a diagonal matrix of size min(m,n)×min(m,n)
	// and V~ is of size n×min(m,n).
	SVDThin

	// SVDNone specifies that no singular vectors should be computed during
	// the decomposition.
	SVDNone SVDKind = 0
)
```

Reply:
We could change the SVD signature to match the original lapack signature. The discussion in the issue concluded supporting all combinations was unnecessary.

Reply:
With the three bits, you get only eight possible combinations with 7 distinct outcomes. Using the bit flags simplifies the API  by making things (largely) orthogonal and the big switch in Factorize become a smaller set of orthogonal conditions with a little bit of preparation before hand.

Reply:
That it true, if your want fine control U-thin/V-full, but that's probably fairly rare. Using three bits gets all the cases above except those two, but for three necessary constants with explicit naming, and one convenience constant for 0 while the approach here requires five constants that use positional implied naming (`SVDNoneThin` really is `SVDNoUThinV`, which would be expressed as `SVDV|SVDThin` in the suggested scheme).

Reply:
See also #756.

Reply:
I think I would be fine with just the current `SVDFull`, `SVDThin`, `SVDNone`, simple, clear and covers a majority of uses. But if we are to refine it, then I slightly prefer the bitfield approach. I agree with @kortschak 's API simplification and orthogonality argument. Also I think the `|` separator makes it easier to see what is being requested, the mixed case constants could look cryptic to an unprepared eye. And if the type were renamed from `SVDKind` to `SVDFlags`, it would better communicate that the parameter is a bitfield. If we keep the way in this PR, then I would include U and V explicitly in the constants, as in `SVDThinUNoneV` or `SVDThinUNoV`.